### PR TITLE
chore : change default chain id to 31337

### PR DESCRIPTION
**Pull request by 0xvv**
Changes discussed in #1431. 

Should the enum in ethers-rs be updated to include 31337 as a second dev chain or to replace the current one ? 
That would also impact the implementation of enable_for_chain_id()

### CODE REVIEW

**Comment by onbjerg** [comment]
This breaks compatibility with DappTools and might break some people's tests. If we want to go for this now, I think we should batch it with some other breaking changes like changing the default `block.number` and `block.timestamp` to 1 instead of 0. wdyt @gakonst?

File: config/src/caching.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ impl StorageCachingConfig {
     /// Whether caching should be enabled for the chain id
     pub fn enable_for_chain_id(&self, chain_id: u64) -> bool {
         // ignore dev chain
-        if chain_id == 1337 {
+        if chain_id == 31337 {
```
Comment (mattsse) [inline config/src/caching.rs:25]: this should be extended to `&[31337, 1337, 99].contains(chain_id)`, because we want to ignore dev chains generally.

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks for this!
can we make this a constant? preferably in the common crate?

**Comment by 0xvv** [comment]
@onbjerg since Dapp appears to be less relevant and the test will not asses parity if the `block.number` and `block.timestamp` change, i suggest renaming DappToolsParityTest to ContractEnvironementTest (or something along those lines), also enables to bring back the chain_id assertion. Or maybe we can turn it into HardhatParityTest ? 

@mattsse to clarify the thing you want as a constant is the dev chain id ?

**Comment by gakonst** [comment]
Looking at our integration tests, DRAI should be safe because it uses [chainId](https://github.com/mds1/drai/blob/main/src/Drai.sol#L100), but it will break RAI because it uses 99, so maybe we should make a PR to RAI tests to use `chainId` instead of hardcoding 99 [here](https://github.com/reflexer-labs/geb/blob/3a86fa290fc9994006914b00aa6fe763d19de73d/src/test/shared/Coin.t.sol#L204).

**Comment by onbjerg** [comment]
> @onbjerg since Dapp appears to be less relevant and the test will not asses parity if the `block.number` and `block.timestamp` change, i suggest renaming `DappToolsParityTest` to `ContractEnvironementTest` (or something along those lines), also enables to bring back the chain_id assertion. Or maybe we can turn it into `HardhatParityTest` ?
> 
> @mattsse to clarify the thing you want as a constant is the dev chain id ?

I think `ContractEnvironmentTest` makes sense, and then adding back the assertions for chain ID and so on that were removed

File: forge/src/lib.rs [hunk 2]
```diff
@@ -79,6 +79,8 @@ pub mod test_helpers {
             gas_limit: 18446744073709551615,
             chain_id: Some(foundry_common::DEV_CHAIN_ID),
             tx_origin: Address::from_str("00a329c0648769a73afac7f9381e08fb43dbea72").unwrap(),
+            block_number: 1,
+            block_timestamp: 1,
```
Comment (onbjerg) [inline forge/src/lib.rs:83]: Also needs to be changed here: https://github.com/foundry-rs/foundry/blob/a97090d6eb087184f2a85965f3e15206b8ba97bc/evm/src/executor/opts.rs#L11

This is just for the tests internally

Comment (0xvv) [reply forge/src/lib.rs:83]: good catch

**Comment by 0xvv** [comment]
LMK what you think about my last commit, is it as an appropriate place to fix dapptools config ?

Should we also change block difficulty to 1 while we are at it ?

File: testdata/core/ContractEnvironment.t.sol [hunk 3]
```diff
@@ -17,15 +17,15 @@ contract DappToolsParityTest is DSTest {
     }
 
     function testEnvironment() public {
-        assertEq(chainId(), 99, "chain id is incorrect");
-        assertEq(block.number, 0);
+        assertEq(chainId(), 31337);
+        assertEq(block.number, 1);
```
Comment (onbjerg) [inline testdata/core/ContractEnvironment.t.sol:21]: smol nit

```suggestion
        assertEq(block.number, 1, "block number is incorrect");
```

File: testdata/core/ContractEnvironment.t.sol [hunk 4]
```diff
@@ -17,15 +17,15 @@ contract DappToolsParityTest is DSTest {
     }
 
     function testEnvironment() public {
-        assertEq(chainId(), 99, "chain id is incorrect");
-        assertEq(block.number, 0);
+        assertEq(chainId(), 31337);
```
Comment (onbjerg) [inline testdata/core/ContractEnvironment.t.sol:20]: smol nit

```suggestion
        assertEq(chainId(), 31337, "chainid is incorrect");
```

**Review by onbjerg — COMMENTED** [review]
otherwise lgtm

**Review by gakonst — APPROVED** [review]
lgtm - very exciting ):

**Comment by peter-ciccolo** [comment]
Just to be clear, why was `block_timestamp` default changed to 1? For parity with hardhat?

**Comment by mds1** [comment]
Using zero is just a bad default because it results in some edge cases. For example when defaulting to zero, you can't multiply it by a value, or subtract 1 from it, which results in unexpected behavior in tests

**Comment by peter-ciccolo** [comment]
OK. Is there a place I should be looking to track this type of breaking change?

**Comment by gakonst** [comment]
I think we'll eventually start publishing clearer changelogs, for now the best bet is tracking the Github. Realize this is not a great answer, so appreciate the understanding.
